### DISCUSSION TITLE
옵셔널 체이닝 연산자에서는 왜 undefined를 반환하는 건가요?

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#39] [궁금증]
옵셔널 체이닝 연산자에서는 왜 undefined를 반환하는 건가요?

**Comment by publdaze** [comment]
- 옵셔널 체이닝 연산자는 깊은 중첩 구조에서 안전하게 프로퍼티에 접근할 수 있도록 설계됨
- 옵셔널 체이닝 연산자 없이 존재하지 않는 프로퍼티에 접근하려면 에러가 발생하고 에러가 발생하지 않도록 하기위해 &&을 사용하는 등 추가 작업이 필요했음
- 즉, 중간 노드가 존재하는지 별도의 확인 과정을 거쳐야 했는데, 옵셔널 체이닝을 통해 undefined를 반환하면서 많은 사례를 처리할 수 있게 됨
- [참고1](https://ko.javascript.info/optional-chaining), [참고2](https://github.com/tc39/proposal-optional-chaining)

**Comment by pipisebastian** [comment]
- 중첩된 객체 구조에서 안전하게 프로퍼티에 접근하고, 중간에 `null` 또는 `undefined`를 만나면 에러를 발생시키지 않고 `undefined`를 반환함으로써 프로그램의 안정성을 높임
